### ISSUE TITLE
FP on S1192 StringLiteralDuplicatedCheck: String literals should not be duplicated produces false positives with struct tags

### ISSUE THREAD

**Issue by hullarb** [SonarSource/sonar-go-archived#337]
Struct tags should be excluded from the duplication checking as they have to be string literals.

For example:
```
struct {
A int `json:",string"`
B int `json:",string"`
C int `json:",string"`
}
```
will be marked as a critical code smell: "Define a constant instead of duplicating this literal 3 times." In go these tags cannot be defined as const.

**Comment by alban-auzeill** [comment]
I can reproduce and I agree this is a false-positive:
```go
type T struct {
  A int `json:",string"`
  B int `json:",string"`
  C int `json:",string"`
}
```
